ipn/ipnlocal/local.go

```diff
@@ -2749,6 +2837,12 @@ func (b *LocalBackend) WatchNotificationsAs(ctx context.Context, actor ipnauth.A
 		cancel:    cancel,
 	}
 	mak.Set(&b.notifyWatchers, sessionID, session)
+	if mask&ipn.NotifyRateLimitNetmaps != 0 {
+		b.setNetmapRateLimit(ipn.DefaultNetmapRateLimit)
+	} else {
+		b.setNetmapRateLimit(0)
+	}
+
```

Comment:
this is kinda weird making a per-session WatchOpt bit affect the LocalBackend globally.

Reply:
What about something like https://github.com/tailscale/tailscale/pull/14120 ?

Reply:
Ack.. Yeah, I'll toss this and run some tests using your patch.  You're correct on the out-of-order comment below.  I just spotted a couple of cases where this gives odd results.